### DISCUSSION TITLE
Issue with pushing large files

### DISCUSSION THREAD

**Discussion by DavidGilson24** [community/community#61221] [Repositories] [answered]
### Select Topic Area

Question

### Body

I've been having an issue where I am trying to push my changes on my repo, but GitHub isn't letting me upload a file more than 100Mb. i have tried to use Git LFS, but to no avail. How can I fix this. The file is a webm file and I can't really split it into smaller files. Here is what im getting:

Uploading LFS objects: 100% (2/2), 237 MB | 2.2 MB/s, done.
Enumerating objects: 25, done.
Counting objects: 100% (25/25), done.
Delta compression using up to 8 threads
Compressing objects: 100% (18/18), done.
Writing objects: 100% (19/19), 226.16 MiB | 2.07 MiB/s, done.
Total 19 (delta 8), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (8/8), completed with 4 local objects.
remote: error: See https://gh.io/lfs for more information.
remote: error: File assets/images/shine_dark.webm is 119.60 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: File assets/images/shine_light.webm is 106.50 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.
To https://github.com/DavidGilson24/Portofolio.git
 ! [remote rejected] main -> main (pre-receive hook declined)
error: failed to push some refs to 'https://github.com/DavidGilson24/Portofolio.git'

**Comment by roma-marshall** [comment]
hello @DavidGilson24, did you install Git LFS? Try this:

```
brew install git-lfs
git lfs track "shine_light.webm"
git add shine_light.webm
git commit -m "Init commit"
git push
```

**Reply by DavidGilson24** [reply, marked as answer]
I have installed git lfs like you described, but I managed to fix the issue. instead of doing git lfs track, I had to do the following:

`git filter-branch --tree-filter 'rm -rf path/to/file' HEAD`

**Reply by shawnnolanjr** [reply]
I've tried this and it did NOT work for me. This has to be the most frustrating thing i have ever run into in the ~15yrs of my dev experience. I have never witnessed something so horrible in my life. This snag of a PITA issue that should easily be solved by running a cmd with a flag, but having to download some other program, or run the filter-branch (which is kind of a sketchy thing rewriting stuff), makes this the most irritating flaw i've ever seen in my life... 

I accidentally uploaded a gif file (which wasn't as large as it says it is), and now i have to wipe out my local env to get it all back? nah, that's f'd up...

**Reply by stephenmontgomery** [reply]
@shawnnolanjr +1 on this sh1tshow - just spent an afternoon on this and still unresolved. Complete lack of transparency, which is supposed to be one the plus points of git.
